File: src/Integrator.js
/* @flow */

export class Integrator {
  target: number;
  value: number;
  k: number;

  constructor(v: number, k: number = 0.2) {
    this.target = v;
    this.value = v;
    this.k = k;
  }

  update(): void {
    this.value += (this.target - this.value) * this.k;
  }
}

File: original/newyears/Integrator.java
public class Integrator
{
  public float target,value,k;

  public Integrator( float v ){
    this(v, 0.2f);
  }

  public Integrator( float v, float k ){
    target = v;
    value = v;
    this.k = k;
  }

  public void update(){
    value += (target-value)*k;
  }
}
